src/handlers/delegatedStakingHandler.ts

```diff
@@ -0,0 +1,190 @@
+    balance: change.balance,
+    rewards: change.rewards,
+    timestamp: change.created_at,
+  });
+  
+  const delegateStakeBalanceChangeResponses = (txs) => txs.map(delegateStakeBalanceChangeResponse);
+
+
+export const delegatedStakes = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  return paginatedQuery(
+    extractPagination(event),
+    toOrdinalCursor,
+    fromOrdinalCursor,
+    {
+      where: {
+        delegate_stake_withdraw_event: null,
+      },
+      orderBy: [{ ordinal: "desc" }],
+    },
+    prisma.delegate_stake_create_events.findMany,
+    delegateStakeCreateResponses
+  );
+};
+  
+  export const delegatedStake = async (event) => {
+    try {
+      const hash = event.pathParameters?.hash;
+      
+      const stake = await prisma.delegate_stake_create_events.findUnique({
+        where: { hash },
+      });
+
+      return respond(stake, delegateStakeCreateResponse);
+    } catch (error) {
+    return handleError(error);
+    }
+  };
+  
+  export const addressDelegatedStakes = async (event) => {
+      const address = event.pathParameters?.address;
+      return paginatedQuery(
+        extractPagination(event),
+        toOrdinalCursor,
+        fromOrdinalCursor,
+        {
+          where: {
+            source_addr: address,
+          },
+          orderBy: [{ ordinal: "desc" }],
+        },
+        prisma.delegate_stake_create_events.findMany,
+        delegateStakeCreateResponses
+      );
+  };
+  
+  export const delegatedStakeWithdrawals = async (event) => {
+    return paginatedQuery(
+        extractPagination(event),
+        toOrdinalCursor,
+        fromOrdinalCursor,
+        {
+          orderBy: [{ ordinal: "desc" }],
+        },
+        prisma.delegate_stake_withdraw_events.findMany,
+        delegateStakeWithdrawResponses
+      );
+  };
+  
+  export const delegatedStakeWithdrawal = async (event) => {
+    try {
+      const hash = event.pathParameters?.hash;
+    
+    const withdrawal = await prisma.delegate_stake_withdraw_events.findUnique({
+        where: { hash },
+    });
+
+    return respond(withdrawal, delegateStakeWithdrawResponse);
+    } catch (error) {
+    return handleError(error);
+    }
+  };
+  
+  export const addressDelegatedStakeWithdrawals = async (event) => {
+    const address = event.pathParameters?.address;
+      return paginatedQuery(
+        extractPagination(event),
+        toOrdinalCursor,
+        fromOrdinalCursor,
+        {
+          where: {
+            source_addr: address,
+          },
+          orderBy: [{ ordinal: "desc" }],
+        },
+        prisma.delegate_stake_withdraw_events.findMany,
+        delegateStakeWithdrawResponse
+      );
+  };
+
+  
+  export const stakingBalanceByAddress = async (
+    event: APIGatewayProxyEvent
+  ): Promise<APIGatewayProxyResult> => {
+    try {
+      const { address } = event.pathParameters || {};
+  
+      const { ordinal } = event.queryStringParameters || {};
```

Comment:
We don't want to support ordinal-based queries for this. Let's only return the latest for now.